contrib/platform/test/com/sun/jna/platform/win32/Kernel32UtilTest.java

```diff
@@ -253,4 +256,34 @@ public final void testWritePrivateProfileSection() throws IOException {
             reader.close();
         }        
     }
+    
+    public void testGetResource() {
+		// On Windows 7, "14" is the type assigned to the "My Computer" icon
+		// (which is named "ICO_MYCOMPUTER")
+    	byte[] results = Kernel32Util.getResource("c:\\windows\\explorer.exe", "14", "ICO_MYCOMPUTER");
```

Comment:
I'll keep that in mind when I clean up the Kernel32 stuff and submit a PR for that DLL.  

Are there any guidelines on things unit tests are *not* allowed to do?

Sent from my iPhone

> On Nov 11, 2015, at 9:41 AM, Lyor Goldstein <notifications@github.com> wrote:
> 
> In contrib/platform/test/com/sun/jna/platform/win32/Kernel32UtilTest.java:
> 
> > @@ -253,4 +256,34 @@ public final void testWritePrivateProfileSection() throws IOException {
> >              reader.close();
> >          }        
> >      }
> > +    
> > +    public void testGetResource() {
> > +		// On Windows 7, "14" is the type assigned to the "My Computer" icon
> > +		// (which is named "ICO_MYCOMPUTER")
> > +    	byte[] results = Kernel32Util.getResource("c:\\windows\\explorer.exe", "14", "ICO_MYCOMPUTER");
> I don't think this will work - I think you can use Kernel32.INSTANCE.GetEnvironmentVariable("WINDIR") and then create the path yourself - e.g.:
> 
> String winDir = Kernel32.INSTANCE.GetEnvironmentVariable("WINDIR");
> ...handle error if any...
> String explorerDir = winDir + File.separator + "explorer.exe";
> // Should check if 'winDir' already ends in File.separatorChar and avoid appending it again if so
> ...run your test...
> —
> Reply to this email directly or view it on GitHub.
>